contracts/btc-light-client/src/state.rs

```diff
@@ -0,0 +1,219 @@
+use crate::error::ContractError;
+use babylon_proto::babylon::btclightclient::v1::BtcHeaderInfo;
+use bitcoin::params::Params;
+use cosmwasm_schema::cw_serde;
+use cosmwasm_std::Order::{Ascending, Descending};
+use cosmwasm_std::{StdError, StdResult, Storage};
+use cw_storage_plus::{Bound, Item, Map};
+use hex::ToHex;
+use prost::Message;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+pub const CONFIG: Item<Config> = Item::new("config");
+
+pub const BTC_BASE_HEADER_HEIGHT: Item<u32> = Item::new("btc_lc_base_header_height");
+pub const BTC_HEADERS: Map<u32, Vec<u8>> = Map::new("btc_lc_headers");
+pub const BTC_HEIGHTS: Map<&[u8], u32> = Map::new("btc_lc_heights");
+pub const BTC_TIP: Item<Vec<u8>> = Item::new("btc_lc_tip");
+
+/// Error type for the state store.
+#[derive(thiserror::Error, Debug, PartialEq)]
+pub enum StoreError {
+    #[error("The bytes cannot be decoded")]
+    Decode(#[from] prost::DecodeError),
+    #[error(transparent)]
+    CosmwasmStd(#[from] StdError),
+    #[error("The BTC height {height} is not found in the storage")]
+    HeightNotFound { height: u32 },
+    #[error("The BTC header with hash {hash} is not found in the storage")]
+    HeaderNotFound { hash: String },
+}
+
```

Comment:
Do these need to be under `error.rs` or here? Same question for other errors

Reply:
Not necessarily, I'd say. I don't think we have to define all the Error types in `error.rs`. It depends on the context. For a large crate-level Error type, we could either:

- Split it into smaller, more focused error types, or
- Encapsulate them hierarchically within relevant modules.

The right choice comes down to maintainability, whichever keeps the codebase clean and easier to reason about. For smaller crate Error, a single error type may suffice, but for larger ones, modularizing errors often helps.